ydb/core/grpc_services/auth_processor/dynamic_node_auth_processor.cpp

```diff
@@ -63,6 +63,38 @@ TVector<std::pair<TString, TString>> X509CertificateReader::ReadSubjectTerms(con
     return extractions;
 }
 
+TVector<std::pair<TString, TString>> X509CertificateReader::ReadAllSubjectTerms(const X509Ptr& x509) {
+    TVector<std::pair<TString, TString>> subjectTerms;
+    X509_NAME* name = X509_get_subject_name(x509.get()); // return internal pointer
+
+    int entryCount = X509_NAME_entry_count(name);
+    subjectTerms.reserve(entryCount);
+    char buf[1024];
+    static const int bufLen = sizeof(buf);
+    for (int i = 0; i < entryCount; i++) {
+        const X509_NAME_ENTRY* entry = X509_NAME_get_entry(name, i);
+        if (!entry) {
+            continue;
+        }
+        const ASN1_STRING* data = X509_NAME_ENTRY_get_data(entry);
+        if (!data) {
+            continue;
+        }
+        int dataLen = (data->length > (bufLen - 1)) ? (bufLen - 1) : data->length;
+        memcpy(buf, data->data, dataLen);
+        buf[dataLen] = '\0';
```

Comment:
why do you need this intermediate buf ?

Reply:
data consists of unsigned char and TString does not have constructor for unsigned char. Remove the buf, use reinterpret_cast<char*> for data